mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/PostgresMailboxFactory.java

```diff
@@ -0,0 +1,43 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.postgres;
+
+import java.util.Optional;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import org.apache.james.backends.postgres.utils.PostgresExecutor;
+import org.apache.james.core.Domain;
+import org.apache.james.mailbox.postgres.mail.dao.PostgresThreadDAO;
+
+public class PostgresMailboxFactory {
+    private final PostgresExecutor.Factory executorFactory;
+
+    @Inject
+    @Singleton
+    public PostgresMailboxFactory(PostgresExecutor.Factory executorFactory) {
+        this.executorFactory = executorFactory;
+    }
+
+    public PostgresThreadDAO createThreadDAO(Optional<Domain> domain) {
+        return new PostgresThreadDAO(executorFactory.create(domain));
+    }
+}
```

Comment:
It is not a `MailboxFactory` if it builds ThreadDAO...

Reply:
Yeah I don't understand that class either... Why you don't follow the current pattern? Should have a Factory subclass in PostgresThreadDAO, to make `PostgresThreadDAO.Factory`, like the other generated factories in `PostgresMailboxSessionMappoerFactory`?

Reply:
I think it would be better to gather all DAOs creation in one feactory instead of one factory per DAO. WDYT?

Reply:
I think the factories being subclasses of their DAO is easier to follow IMO